xray/plot/plot.py

```diff
@@ -452,6 +452,8 @@ def newplotfunc(darray, ax=None, xincrease=None, yincrease=None,
             # pcolormesh
             kwargs['extend'] = cmap_params['extend']
             kwargs['levels'] = cmap_params['levels']
+            if 'colors' in kwargs:
```

Comment:
I'm fine with this.  It will just be important to distinguish between a list of colors and a ListedColorMap.